src/hapticfeedback.h

```diff
@@ -0,0 +1,62 @@
+/* antimicrox Gamepad to KB+M event mapper
+ * Copyright (C) 2022 Max Maisel <max.maisel@posteo.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <QObject>
+#include <QString>
+#include <SDL2/SDL_gamecontroller.h>
+#include <stdint.h>
+
+#include "hapticfeedbackmode.h"
+
+/**
+ * @brief Represents a single haptic feedback effect on a PS5 controller trigger.
+ */
+class HapticFeedback : QObject
```

Comment:
If a class represents only a single haptic feedback for a single controller, then it shouldn't have such a generic name. It should be a bit more descriptive like `HapticTriggerPS5` if you think this class can be applicable only for this purpose or something like`HapticControllerPS5` if you think it can also have (or easily implement) other haptic-related features for PS5 Dualsense

Reply:
I renamed it to HapticTriggerPs5 so a HapticTrigger base class can be added later when there are more controllers with haptioc triggers.